src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcObjectMethodExecutor.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Diagnostics;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public struct ActionResultOfTUnwrapper
+    {
+        public ActionResultOfTUnwrapper(Type returnType)
```

Comment:
Consider just putting this on ActionResult<T>. Like, declare a new interface `IUnwrapActionResult` and then use that.